ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java

```diff
@@ -811,6 +838,12 @@ public InputSplit[] getSplits(JobConf job, int numSplits) throws IOException {
           pushDownProjection = true;
           // push down filters and as of information
           pushFiltersAndAsOf(newjob, tableScan, this.mrwork);
+          final List<String> groupingPartitionColumns = tableScan.getConf().getGroupingPartitionColumns();
```

Comment:
could we extract this into a helper method like `addGroupingDetails()`

Reply:
Done
https://github.com/apache/hive/pull/5409/commits/ae7444e8e1c2de95bfc1915af3d3c0245af4b472